### ISSUE TITLE
NaN during segmentation (beginner user)

### ISSUE THREAD

**Issue by xShGuardian** [open-mmlab/mmsegmentation#3493]
Hello, everyone! I'm trying to segment retinal vessels with the help of mmsegmentation. My images are 512x512x3 (already preprocessed and normalized)  , while my masks are 512x512 (0-1). In training, I get the following problem: 

12/27 23:20:27 - mmengine - INFO - Iter(train) [ 105/2000]  lr: 5.0000e-05  eta: 0:51:13  time: 1.2916  data_time: 0.4326  memory: 10578  grad_norm: 0.7342  loss: 0.0441  decode.loss_ce: 0.0147  decode.acc_seg: 100.0000  aux.loss_ce: 0.0295  aux.acc_seg: 100.0000
12/27 23:24:31 - mmengine - INFO - per class results:
12/27 23:24:31 - mmengine - INFO - 
+------------+-------+-------+
|   Class    |  IoU  |  Acc  |
+------------+-------+-------+
| background | 100.0 | 100.0 |
|   vessel   |  nan  |  nan  |
+------------+-------+-------+
12/27 23:24:31 - mmengine - INFO - Iter(val) [120/120]    aAcc: 100.0000  mIoU: 100.0000  mAcc: 100.0000  data_time: 1.1589  time: 1.3398_

**I don't get why i'm getting NaNs**. 
I tried with the following RESNET config:

compile = False
crop_size = (
    512,
    512,
)
data_root = '/content/drive/MyDrive/Colab Notebooks/Splitted_Database'
dataset_type = 'RetinalDataset'
default_hooks = dict(
    checkpoint=dict(by_epoch=False, interval=1000, type='CheckpointHook'),
    logger=dict(interval=500, log_metric_by_epoch=False, type='LoggerHook'),
    param_scheduler=dict(type='ParamSchedulerHook'),
    sampler_seed=dict(type='DistSamplerSeedHook'),
    timer=dict(type='IterTimerHook'),
    visualization=dict(draw=True, interval=500, type='SegVisualizationHook'))
default_scope = 'mmseg'
env_cfg = dict(
    cudnn_benchmark=True,
    dist_cfg=dict(backend='nccl'),
    mp_cfg=dict(mp_start_method='fork', opencv_num_threads=0))
img_ratios = [
    0.5,
    0.75,
    1.0,
    1.25,
    1.5,
    1.75,
]
log_level = 'INFO'
log_processor = dict(by_epoch=False)
model = dict(
    auxiliary_head=dict(
        align_corners=False,
        channels=256,
        concat_input=False,
        dropout_ratio=0.1,
        ignore_index=255,
        in_channels=1024,
        in_index=2,
        loss_decode=dict(
            loss_weight=0.4, type='CrossEntropyLoss', use_sigmoid=False),
        norm_cfg=dict(requires_grad=True, type='BN'),
        num_classes=2,
        num_convs=1,
        type='FCNHead'),
    backbone=dict(
        contract_dilation=True,
        depth=50,
        dilations=(
            1,
            1,
            2,
            4,
        ),
        norm_cfg=dict(requires_grad=True, type='BN'),
        norm_eval=False,
        num_stages=4,
        out_indices=(
            0,
            1,
            2,
            3,
        ),
        strides=(
            1,
            2,
            1,
            1,
        ),
        style='pytorch',
        type='ResNetV1c'),
    data_preprocessor=dict(
        bgr_to_rgb=True,
        mean=[
            0,
            0,
            0,
        ],
        pad_val=255,
        seg_pad_val=255,
        size=(
            512,
            512,
        ),
        std=[
            1,
            1,
            1,
        ],
        type='SegDataPreProcessor'),
    decode_head=dict(
        align_corners=False,
        channels=512,
        dropout_ratio=0.1,
        ignore_index=255,
        in_channels=2048,
        in_index=3,
        loss_decode=[
            dict(loss_weight=1.0, type='CrossEntropyLoss', use_sigmoid=False),
        ],
        norm_cfg=dict(requires_grad=True, type='BN'),
        num_classes=2,
        pool_scales=(
            1,
            2,
            3,
            6,
        ),
        type='PSPHead'),
    pretrained='open-mmlab://resnet50_v1c',
    test_cfg=dict(mode='whole'),
    train_cfg=dict(),
    type='EncoderDecoder')
norm_cfg = dict(requires_grad=True, type='BN')
optim_wrapper = dict(
    clip_grad=dict(max_norm=1, norm_type=2),
    optimizer=dict(
        betas=(
            0.9,
            0.999,
        ), lr=5e-05, type='AdamW', weight_decay=0.01),
    type='AmpOptimWrapper')
optimizer = dict(
    betas=(
        0.9,
        0.999,
    ), lr=5e-05, type='AdamW', weight_decay=0.01)
param_scheduler = dict(
    begin=2000,
    by_epoch=False,
    end=10000,
    gamma=0.1,
    milestones=[
        6000,
        8000,
    ],
    type='MultiStepLR')
randomness = dict(seed=0)
resume = False
save_dir = '/content/drive/MyDrive/Colab Notebooks/results'
test_cfg = dict(type='TestLoop')
test_dataloader = dict(
    batch_size=1,
    dataset=dict(
        data_prefix=dict(
            img_path='Original/test', seg_map_path='Ground truth/test'),
        data_root=
       '/content/drive/MyDrive/Colab Notebooks/Splitted_Database',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='LoadAnnotations'),
            dict(type='PackSegInputs'),
        ],
        type='RetinalDataset'),
    num_workers=1,
    persistent_workers=True,
    sampler=dict(shuffle=False, type='DefaultSampler'))
test_evaluator = dict(
    ignore_index=255, iou_metrics=[
        'mIoU',
    ], type='IoUMetric')
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations'),
    dict(type='PackSegInputs'),
]
train_cfg = dict(max_iters=2000, type='IterBasedTrainLoop', val_interval=200)
train_dataloader = dict(
    batch_size=4,
    dataset=dict(
        data_prefix=dict(
            img_path='Original/train', seg_map_path='Ground truth/train'),
        data_root=
        '/content/drive/MyDrive/Colab Notebooks/Splitted_Database',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='LoadAnnotations'),
            dict(prob=0.5, type='RandomFlip'),
            dict(degree=(
                -15.0,
                15.0,
            ), prob=0.5, type='RandomRotate'),
            dict(type='PackSegInputs'),
        ],
        type='RetinalDataset'),
    num_workers=2,
    persistent_workers=True,
    sampler=dict(shuffle=True, type='InfiniteSampler'))
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations'),
    dict(prob=0.5, type='RandomFlip'),
    dict(degree=(
        -15.0,
        15.0,
    ), prob=0.5, type='RandomRotate'),
    dict(type='PackSegInputs'),
]
tta_model = dict(type='SegTTAModel')
tta_pipeline = [
    dict(backend_args=None, type='LoadImageFromFile'),
    dict(
        transforms=[
            [
                dict(keep_ratio=True, scale_factor=0.5, type='Resize'),
                dict(keep_ratio=True, scale_factor=0.75, type='Resize'),
                dict(keep_ratio=True, scale_factor=1.0, type='Resize'),
                dict(keep_ratio=True, scale_factor=1.25, type='Resize'),
                dict(keep_ratio=True, scale_factor=1.5, type='Resize'),
                dict(keep_ratio=True, scale_factor=1.75, type='Resize'),
            ],
            [
                dict(direction='horizontal', prob=0.0, type='RandomFlip'),
                dict(direction='horizontal', prob=1.0, type='RandomFlip'),
            ],
            [
                dict(type='LoadAnnotations'),
            ],
            [
                dict(type='PackSegInputs'),
            ],
        ],
        type='TestTimeAug'),
]
val_cfg = dict(type='ValLoop')
val_dataloader = dict(
    batch_size=1,
    dataset=dict(
        data_prefix=dict(
            img_path='Original/validation',
            seg_map_path='Ground truth/validation'),
        data_root=
       '/content/drive/MyDrive/Colab Notebooks/Splitted_Database',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='LoadAnnotations'),
            dict(type='PackSegInputs'),
        ],
        type='RetinalDataset'),
    num_workers=1,
    persistent_workers=True,
    sampler=dict(shuffle=False, type='DefaultSampler'))
val_evaluator = dict(
    ignore_index=255, iou_metrics=[
        'mIoU',
    ], type='IoUMetric')
visualizer = dict(
    classes=[
        'background',
        'vessel',
    ],
    dataset_name='RetinalDataset',
    name='visualizer',
    palette=[
        (
            0,
            0,
            0,
        ),
        (
            255,
            255,
            255,
        ),
    ],
    save_dir=
    '/content/drive/MyDrive/Colab Notebooks/results',
    type='SegLocalVisualizer',
    vis_backends=[
        dict(type='LocalVisBackend'),
        dict(type='TensorboardVisBackend'),
    ])
work_dir = '/content/drive/MyDrive/Colab Notebooks/results'

I don't know what to do at this point. Any idea?

**Comment by innavoig23** [comment]
How did you solve that?
